### PR TITLE
[XLA:GPU] Fix the bug of thread conflicts for running command buffer cublasLtCmd

### PR DESCRIPTION
**Pull request by shawnwang18**
It is a bug that CublasLtCmd's save the gemm plan and algorithm across Initialize and Execute.

### CODE REVIEW

**Review by ezhulenev — APPROVED** [review]

**Comment by penpornk** [comment]
Would it be possible to add a test (or update existing tests) for this, e.g., a unit test showing that the values changed because they aren't cached?
We have a new policy requiring test updates for any PRs that cause behavioral changes.

**Review by ezhulenev — APPROVED** [review]

File: xla/service/gpu/runtime/command_buffer_cmd.h [hunk 1]
```diff
@@ -777,13 +777,11 @@ class CublasLtCmd : public TracedCommandBufferCmd {
                       se::gpu::BlasLt::MatmulPlanPtr>
       matmul_plans_cache_;
 
+  absl::Mutex matmul_algorithm_cache_mutex_;
```
Comment (cota) [inline xla/service/gpu/runtime/command_buffer_cmd.h:780]: Leftover? If so, please remove.

File: xla/service/gpu/runtime/command_buffer_thunk_test.cc [hunk 2]
```diff
@@ -673,57 +646,89 @@ TEST(CommandBufferThunkTest, CublasLtCmd) {
...
+    TF_ASSERT_OK(stream->BlockHostUntilDone());
 
-  ASSERT_EQ(dst, std::vector<float>({11, 11, 11, 27, 27, 27}));
+    // Copy `out` data back to host.
+    std::vector<float> dst(6, 0);
+    TF_ASSERT_OK(stream->Memcpy(dst.data(), d, d_length));
 
-  // Try to update the command buffer with the same buffers.
-  TF_ASSERT_OK(stream->MemZero(&updated_d, d_length));
+    // ASSERT_EQ(dst, std::vector<float>({11, 11, 11, 27, 27, 27}));
```
Comment (cota) [inline xla/service/gpu/runtime/command_buffer_thunk_test.cc:692]: Why leave this commented out?

**Review by ezhulenev — APPROVED** [review]

**Review by cota — APPROVED** [review]
Thanks!

**Pull request closed without merging**